ocs_ci/deployment/deployment.py

```diff
@@ -561,6 +561,11 @@ def deploy_ocs_via_operator(self):
                         }
                     cluster_data['spec']['resources'] = resources
 
+            cluster_data['spec']['resources']['noobaa-endpoint'] = {
```

Comment:
Elad, this block should be moved to line 542.

I would change the logic: https://github.com/red-hat-storage/ocs-ci/pull/3153/files#diff-8571fc52f1040ac2cb01985da6d61cc93a535f812814c200b45d00daf28c2e3fR538-R539 here to:

```python
                cluster_data['spec']['resources'] = {
                    resource: deepcopy(none_resources) for resource in resources
                }
                if ocs_version >= 4.5:
                    cluster_data['spec']['resources']['noobaa-endpoint'] = {
                        'limits': {'cpu': 1, 'memory': '500Mi'},
                        'requests': {'cpu': 1, 'memory': '500Mi'}
                    }
```

Reply:
@petr-balogh won't this change be relevant for clusters that don't fall under `allow_lower_instance_requirements`? 
The intention is to add constant values to the resource requests and limits, right?

Reply:
IIUC the issue discussed on our last mtg with @nb-ohad was related only on clusters deployed with lower resources as only if that's true we allow and run this block of the code:

https://github.com/red-hat-storage/ocs-ci/blob/b996cf9ab9cad58007255f3d603be1c7dd5665ac/ocs_ci/deployment/deployment.py#L532-L542

Which will set resources to empty dirs.  In all other cases we are just following documentation and setting up resources/limits based on the documentation like for example LSO deployment on AWS I3.

In any other case we should not set the limits - so the default from the product will be used and that should be OK.

Reply:
@ebenahar @petr-balogh 
The main takeaway from our mtg was that the values for the resources configuration (requests and limits) should always be set and should never be `null` or `{}`, at least for the noobaa-endpoint configuration, if not for all.

Now, I am not sure where this code should lay as I am not aware of all the details and the code structure of the deployment code.

Reply:
Yep, and this should happen in our code only in the block mentioned above :

```python
if config.DEPLOYMENT.get('allow_lower_instance_requirements'): 
```

For now, we will start with the noobaa endpoint as it will take more time to find the proper configuration for all other resources.

Reply:
addressed now